File: 6-python/sentimental-cash/cash.py
# Cash in Python

"""
Note:
I decided not to use the get_float training wheel from cs50 module.
It's because I'm comfortable with Python's int and float data type.
"""

while True:
    try:
        # Ask how many dollars the customer is owed
        dollars = float(input("Change owed: "))

        # The next lines are my arithmetic logic to compute the...
        # minimum number of coins required to give a user change.

        # Calculate the number of quarters to give the customer
        quarters = int(dollars * 100) // 25

        # Calculate the number of dimes to give the customer
        dimes = int(dollars * 100) % 25 // 10

        # Calculate the number of nickels to give the customer
        nickels = int(dollars * 100) % 25 % 10 // 5

        # Calculate the number of pennies to give the customer
        pennies = int(dollars * 100) % 25 % 10 % 5

        if dollars < 0:
            # Reprompt
            continue
        elif dollars == "" or dollars == " ":
            # Reprompt
            continue
    except ValueError:
        # Reprompt if user inputs a non-negative value
        pass
    else:
        # Print total number of dollars to give the customer
        give_dollars = quarters + dimes + nickels + pennies
        print(give_dollars)
        break

File: 1-c/cash/cash.c
/* Program that prompts user for the number of cents...
 * that a customer is owed and then prints the smallest number...
 * of coins with which that change can be made
 */

#include <cs50.h>
#include <stdio.h>

// Function prototypes
int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("%i\n", coins);
}

// Function definitions
int get_cents(void)
{
    // TODO
    int cents;

    do
    {
        cents = get_int("Changed owed: ");
    }
    while (cents < 0);

    return cents;
}

int calculate_quarters(int cents)
{
    // TODO
    int quarters = 0;

    if (cents >= 25)
    {
        quarters = cents / 25;
    }
    return quarters;
}

int calculate_dimes(int cents)
{
    // TODO
    int dimes = 0;

    if (cents >= 10)
    {
        dimes = cents / 10;
    }
    return dimes;
}

int calculate_nickels(int cents)
{
    // TODO
    int nickels = 0;

    if (cents >= 5)
    {
        nickels = cents / 5;
    }
    return nickels;
}

int calculate_pennies(int cents)
{
    // TODO
    return cents;
}